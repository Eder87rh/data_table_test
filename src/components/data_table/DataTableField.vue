<template>
    <td>
        <div v-if="!editing" class="data-table-field-container">
            <div>
                {{ this.data }}
            </div>
            <i v-if="this.editable"
                class="material-icons md-18 md-light editable-icon"
                @click="editing = !editing"
            >
                edit
            </i>
        </div>
        <div v-else
            class="data-table-field-container-vertical"
        >
            <textarea v-model="value" class="textarea"/>
            <button class="button"
                @click="saveNewValue"
            >
                SAVE
            </button>
        </div>
    </td>
</template>

<script>
export default {
    props:{
        data:{
            required:true,
            type:String
        },
        index:{
            required:true,
            type:Number,
        },
        editable:{
            default:false,
            type:Boolean
        }
    },
    data(){
        return {
            editing:false,
            value:''
        }
    },
    mounted(){
        this.value = this.data
    },
    methods:{
        saveNewValue(){
            this.$store.commit('updateDescription', {
                index: this.index,
                description: this.value
            })
            this.editing = false
        }
    }
}
</script>

<style lang="scss" scoped>
    .data-table-field-container{
        display:flex;
    }
    .data-table-field-container-vertical{
        display:flex;
        flex-direction: column;
    }
    .editable-icon{
        margin-left:$s-size;
        cursor: pointer;
    }
    .textarea{
        min-height: 8.2rem;
    }
</style>
