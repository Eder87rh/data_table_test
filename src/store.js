import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    data:[
      {
        id: '3471DA17-401F-9633-BF81-4CADA6FD5C79',
        name: 'Kyra Lester',
        description: 'Curabitur dictum. Phasellus in',
        date: '2017-07-23T04:24:49-07:00',
        amount: '345.54'
      },
      {
          id: '3471DA17-401F-9633-BF81-4CADA6FD5C79',
          name: 'Kyra Lester',
          description: 'Curabitur dictum. Phasellus in',
          date: '2017-07-23T04:24:49-07:00',
          amount: '345.54'
      }
    ]
  },
  mutations: {

  },
  actions: {
    
  }
})
