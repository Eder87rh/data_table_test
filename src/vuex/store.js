import Vue from 'vue'
import Vuex from 'vuex'
import modPayments from './modules/payments'
import moodFilters from './modules/filters'

Vue.use(Vuex)

export default new Vuex.Store({
  modules:{
    modPayments,
    moodFilters,
  },
  state: {
    
  },
  mutations: {

  },
  actions: {

  }
})
