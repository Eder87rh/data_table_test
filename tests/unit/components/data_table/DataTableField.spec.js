import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import DataTableField from '@/components/data_table/DataTableField.vue'
import store from '@/vuex/store.js'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('DataTableField.vue',() => {
    it('match snapshot normal field',() => {
        const wraper = shallowMount(DataTableField, {
            mocks: {
                $store: store
            },
            propsData:{
                data:'Test',
                index:0,
                editable:false
            },
            localVue
        })
        expect(wraper.html()).toMatchSnapshot()
    })

    it('match snapshot editable field',() => {
        const wraper = shallowMount(DataTableField, {
            mocks: {
                $store: store
            },
            propsData:{
                data:'Editable',
                index:0,
                editable:true
            },
            localVue
        })
        expect(wraper.html()).toMatchSnapshot()
    })
})