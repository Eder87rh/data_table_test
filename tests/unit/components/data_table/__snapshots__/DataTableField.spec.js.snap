// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTableField.vue match snapshot editable field 1`] = `
<td>
  <div class="data-table-field-container">
    <div>
      Editable
    </div> <i class="material-icons md-18 md-light editable-icon">
      edit
    </i>
  </div>
</td>
`;

exports[`DataTableField.vue match snapshot normal field 1`] = `
<td>
  <div class="data-table-field-container">
    <div>
      Test
    </div>
    <!---->
  </div>
</td>
`;
